rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbCopyKeysNode.java

```diff
@@ -21,89 +21,101 @@
 import org.thingsboard.common.util.JacksonUtil;
 import org.thingsboard.rule.engine.api.RuleNode;
 import org.thingsboard.rule.engine.api.TbContext;
-import org.thingsboard.rule.engine.api.TbNode;
 import org.thingsboard.rule.engine.api.TbNodeConfiguration;
 import org.thingsboard.rule.engine.api.TbNodeException;
 import org.thingsboard.rule.engine.api.util.TbNodeUtils;
+import org.thingsboard.rule.engine.util.TbMsgSource;
 import org.thingsboard.server.common.data.plugin.ComponentType;
 import org.thingsboard.server.common.msg.TbMsg;
-import org.thingsboard.server.common.msg.TbMsgMetaData;
 
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 @Slf4j
 @RuleNode(
         type = ComponentType.TRANSFORMATION,
-        name = "copy keys",
+        name = "copy key-values",
+        version = 1,
         configClazz = TbCopyKeysNodeConfiguration.class,
-        nodeDescription = "Copies the msg or metadata keys with specified key names selected in the list",
-        nodeDetails = "Will fetch fields values specified in list. If specified field is not part of msg or metadata fields it will be ignored." +
-                "Returns transformed messages via <code>Success</code> chain",
+        nodeDescription = "Copies key-values from message to message metadata or vice-versa.",
+        nodeDetails = "Fetches key-values from message or message metadata based on the keys list specified in the configuration " +
+                "and copies them into message metadata or into message itself in accordance with the fetch source. " +
+                "Keys that are absent in the fetch source will be ignored. " +
+                "Use regular expression(s) as a key(s) to copy keys by pattern.<br><br>" +
+                "Output connections: <code>Success</code>, <code>Failure</code>.",
```

Comment:
Copies specified key-value pairs from the message body to metadata, or vice-versa, according to the configured direction and keys. Regular expressions can be used to define which keys-value pairs to copy. Any configured key not found in the source will be ignored.

Output connections:
Success - The node successfully processed and forwarded the message.
Failure - The node encountered an issue processing the message.